### ISSUE TITLE
`dep ensure` changes Gopkg.lock when restoring dependencies

### ISSUE THREAD

**Issue by samuelkarp** [samuelkarp/amazon-ecr-containerd-resolver#2]
`dep ensure` changes the Gopkg.lock unexpectedly during restore, causing CI to fail.

[This build](https://travis-ci.org/samuelkarp/amazon-ecr-containerd-resolver/builds/247754884) failed.  When I removed my `vendor` directory locally and ran `make vendor`, I got this diff unexpectedly:

```diff
diff --git a/Gopkg.lock b/Gopkg.lock
index 7db50dc..c50c6af 100644
--- a/Gopkg.lock
+++ b/Gopkg.lock
@@ -24,9 +24,10 @@
   revision = "0d9f544bb94aac29c185968483459ef74d6deb5d"
 
 [[projects]]
+  branch = "master"
   name = "github.com/containerd/containerd"
-  packages = [".","api/services/containers/v1","api/services/content/v1","api/services/diff/v1","api/services/events/v1","api/services/images/v1","api/services/namespaces/v1","api/services/snapshot/v1","api/services/tasks/v1","api/services/version/v1","api/types/descriptor","api/types/event","api/types/mount","api/types/task","containers","content","events","images","log","metadata","mount","namespaces","plugin","reference","remotes","remotes/docker","remotes/docker/schema1","rootfs","services/content","services/diff","services/images","services/snapshot","snapshot"]
-  revision = "c4da4ed3932fbe8e35f43d6495731c56e9e1052c"
+  packages = [".","api/services/containers/v1","api/services/content/v1","api/services/diff/v1","api/services/events/v1","api/services/images/v1","api/services/namespaces/v1","api/services/snapshot/v1","api/services/tasks/v1","api/services/version/v1","api/types","api/types/task","containers","content","events","identifiers","images","log","metadata","mount","namespaces","plugin","reference","remotes","remotes/docker","remotes/docker/schema1","rootfs","services/content","services/diff","services/images","services/snapshot","snapshot"]
+  revision = "7ec1497a6bbe76f30f7ddbda403610b40ff8edb3"
 
 [[projects]]
   branch = "master"
@@ -141,6 +142,6 @@
 [solve-meta]
   analyzer-name = "dep"
   analyzer-version = 1
-  inputs-digest = "2677790afbfe3cc8671e35892291786d4ae4448eecaa9cb4a9f49a3b50bb98c0"
+  inputs-digest = "aac69bc033e8ad1fb360365a370641b0ed503bdec10c2fb05e3c1c3a40b20b1c"
   solver-name = "gps-cdcl"
   solver-version = 1

```

Fixed for now with https://github.com/samuelkarp/amazon-ecr-containerd-resolver/commit/ef051e068ba1a973323cf5ab00283c36a569b223 (forcing a revision of `github.com/opencontainers/runtime-spec` instead of `github.com/containerd/containerd` because I'd like this to keep up with containerd `master`), but I need to find a better solution.

**Comment by samuelkarp** [comment]
Looks like this should now be solved with [v0.3.1](https://github.com/golang/dep/releases/tag/v0.3.1), specifically using `dep ensure -vendor-only`.

**Comment by samuelkarp** [comment]
This is resolved with https://github.com/samuelkarp/amazon-ecr-containerd-resolver/commit/26eb67769be8de02fe9218635947f38ddacbb853.
